File: BasicCasting/src/main/java/com/bitmovin/player/samples/casting/basic/PlayerActivity.java
/*
 * Bitmovin Player Android SDK
 * Copyright (C) 2017, Bitmovin GmbH, All Rights Reserved
 *
 * This source code and its use and distribution, is subject to the terms
 * and conditions of the applicable license agreement.
 */

package com.bitmovin.player.samples.casting.basic;

import android.os.Bundle;
import android.view.Window;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.core.view.WindowInsetsControllerCompat;

import com.bitmovin.player.PlayerView;
import com.bitmovin.player.api.Player;
import com.bitmovin.player.api.source.SourceConfig;
import com.bitmovin.player.casting.BitmovinCastManager;

public class PlayerActivity extends AppCompatActivity {
    public static final String SOURCE_URL = "SOURCE_URL";
    public static final String SOURCE_TITLE = "SOURCE_TITLE";

    private PlayerView playerView;
    private Player player;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        EdgeToEdge.enable(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_player);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.root), (view, insets) -> {
            Insets bars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            view.setPadding(bars.left, bars.top, bars.right, bars.bottom);
            return WindowInsetsCompat.CONSUMED;
        });
        Window window = getWindow();
        WindowInsetsControllerCompat insetsController = WindowCompat.getInsetsController(window,
                window.getDecorView());
        insetsController.setAppearanceLightStatusBars(true);
        insetsController.setAppearanceLightNavigationBars(true);

        // Update the context the BitmovinCastManager is using
        // This should be done in every Activity's onCreate using the cast function
        BitmovinCastManager.getInstance().updateContext(this);

        String sourceUrl = getIntent().getStringExtra(SOURCE_URL);
        String sourceTitle = getIntent().getStringExtra(SOURCE_TITLE);
        if (sourceUrl == null || sourceTitle == null) {
            finish();
        }

        this.playerView = this.findViewById(R.id.bitmovinPlayerView);
        this.player = this.playerView.getPlayer();

        this.initializePlayer(sourceUrl, sourceTitle);
    }

    @Override
    protected void onStart() {
        this.playerView.onStart();
        super.onStart();
    }

    @Override
    protected void onResume() {
        super.onResume();
        this.playerView.onResume();
    }

    @Override
    protected void onPause() {
        this.playerView.onPause();
        super.onPause();
    }

    @Override
    protected void onStop() {
        this.playerView.onStop();
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        this.playerView.onDestroy();
        super.onDestroy();
    }

    protected void initializePlayer(String sourceUrl, String sourceTitle) {

        // Create a new source item
        SourceConfig sourceItem = SourceConfig.fromUrl(sourceUrl);
        sourceItem.setTitle(sourceTitle);


        // load source using the created source item
        this.player.load(sourceItem);
    }
}

File: BasicCastingKotlin/src/main/java/com/bitmovin/player/samples/casting/basic/PlayerActivity.kt
package com.bitmovin.player.samples.casting.basic

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.updatePadding
import com.bitmovin.player.api.Player
import com.bitmovin.player.api.source.SourceConfig
import com.bitmovin.player.casting.BitmovinCastManager
import com.bitmovin.player.samples.casting.basic.databinding.ActivityPlayerBinding

const val SOURCE_URL = "SOURCE_URL"
const val SOURCE_TITLE = "SOURCE_TITLE"

class PlayerActivity : AppCompatActivity() {
    private lateinit var player: Player
    private lateinit var binding: ActivityPlayerBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        enableEdgeToEdge()
        super.onCreate(savedInstanceState)
        binding = ActivityPlayerBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(binding.root) { view, windowInsets ->
            val insets = windowInsets.getInsets(WindowInsetsCompat.Type.systemBars())
            view.updatePadding(insets.left, insets.top, insets.right, insets.bottom)
            WindowInsetsCompat.CONSUMED
        }
        WindowCompat.getInsetsController(window, window.decorView).apply {
            isAppearanceLightStatusBars = true
            isAppearanceLightNavigationBars = true
        }

        // Update the context the BitmovinCastManager is using
        // This should be done in every Activity's onCreate using the cast function
        BitmovinCastManager.getInstance().updateContext(this)

        val sourceUrl = intent.getStringExtra(SOURCE_URL)
        val sourceTitle = intent.getStringExtra(SOURCE_TITLE)
        if (sourceUrl == null || sourceTitle == null) {
            finish()
            return
        }

        player = binding.bitmovinPlayerView.player!!

        initializePlayer(sourceUrl, sourceTitle)
    }

    override fun onStart() {
        binding.bitmovinPlayerView.onStart()
        super.onStart()
    }

    override fun onResume() {
        super.onResume()
        binding.bitmovinPlayerView.onResume()
    }

    override fun onPause() {
        binding.bitmovinPlayerView.onPause()
        super.onPause()
    }

    override fun onStop() {
        binding.bitmovinPlayerView.onStop()
        super.onStop()
    }

    override fun onDestroy() {
        binding.bitmovinPlayerView.onDestroy()
        super.onDestroy()
    }

    private fun initializePlayer(sourceUrl: String, sourceTitle: String) {
        // Create a new source item
        val sourceItem = SourceConfig.fromUrl(sourceUrl).apply {
            title = sourceTitle
        }

        // load source using the created source item
        player.load(sourceItem)
    }
}
